radixdlt/babylon-wallet-android
# ABW-1842 - Root detection implemented based on rootbeer lib.

## Description
This PR adds root detection in the wallet.
The root detection check is triggered whenever you launch the app.

## How to test
In order to test it you obviously need a root device which might be difficult, however I can provide a way how to achieve root emulator on Android Studio.
- Create emulator via Device Manager.
- Run newly created device (you might also add some screen security (pin, swipe, etc))
- Run command -> adb root
- You should see **restarting adbd as root** in the terminal. It means your emulator device is rooted.

- Now launch the app (fresh or already existing)
- Verify root check dialog is shown 
- Verify that if you acknowledge you won't see it next time unless you uninstall the app

## Video
https://github.com/radixdlt/babylon-wallet-android/assets/108684750/a1dba4e7-f46e-4a21-b7cb-b2450110f57d

## PR submission checklist
- [x] I have verified that root detection dialog is shown on rooted device
